### PR TITLE
upstream: fix the header file path typo

### PR DESCRIPTION
**Pull request by stevenzzzz**
Description: Fix the header path error introudced in pr #8166 

Risk Level: LOW
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: source/common/upstream/health_checker_impl.h [hunk 1]
```diff
@@ -11,7 +12,6 @@
 #include "common/stream_info/stream_info_impl.h"
 #include "common/upstream/health_checker_base_impl.h"
 
-#include "include/envoy/api/_virtual_includes/api_interface/envoy/api/api.h"
```
Comment (PiotrSikora) [inline source/common/upstream/health_checker_impl.h]: It might be worth adding a check against `_virtual_includes` to `./tools/check_format.py`.

Comment (alyssawilk) [reply source/common/upstream/health_checker_impl.h]: Yeah, can you pick this up as a follow-up?

Comment (stevenzzzz) [reply source/common/upstream/health_checker_impl.h]: I added a regex check there. Wording could probably be better though.

**Pull request merged**